### PR TITLE
Skip allowed list check for scratch org plans.

### PR DESCRIPTION
**Pull request by jgerigmeyer**
## Cute animal pic
_Because everyone likes pictures of [animals](https://unsplash.com/t/animals)._

## Link to Trello card (if applicable)
_If this PR relates to a Trello card, don't forget to reference this PR on the card as well._

- https://trello.com/c/V8FwiNnF

## Description
_The commit messages say what you did; this should explain why and/or how._
- [x] Description is in Trello card

### CODE REVIEW

**Comment by jgerigmeyer** [comment]
[Ignore allowed_list checks for scratch-org plans.](https://trello.com/c/V8FwiNnF/172-ignore-allowedlist-checks-for-scratch-org-plans)

File: metadeploy/api/admin.py [hunk 1]
```diff
@@ -156,8 +159,27 @@ class PlanTemplateAdmin(MetadeployTranslatableAdmin):
...
+        cleaned_data = super().clean()
+        visible_to = cleaned_data.get("visible_to")
+        supported_orgs = cleaned_data.get("supported_orgs")
+        if visible_to and supported_orgs != SUPPORTED_ORG_TYPES.Persistent:
+            raise forms.ValidationError(
+                _(
+                    'Restricted plans (with a "visible to" AllowedList) can only support persistent org types.'
+                )
+            )
+        return cleaned_data
```
Comment (davisagli) [inline metadeploy/api/admin.py:177]: Is this the right place for this validation? It's performed for changes through the admin UI, but not via the admin REST API.

Comment (jerivas) [reply metadeploy/api/admin.py:177]: Maybe we can raise `ValidationError` on the model's `clean` method (which would work out of the box in the admin I think) and make sure it's called by the viewset's create and update methods, where it's translated to something DRF can re-surface.

Comment (jgerigmeyer) [reply metadeploy/api/admin.py]: @davisagli (cc @jerivas) I moved this to the model `clean` method, which works similarly when changed via the admin.

The `PlanViewSet` inherits from `ReadOnlyModelViewSet`, so there is no support practically for any create/update actions via the DRF API as far as I can tell. So I did **not** add validation to the `PlanSerializer`, but I'm open to doing that if you'd prefer.

**Review by jerivas — APPROVED** [review]

**Review by davisagli — APPROVED** [review]

File: metadeploy/api/admin.py (hunk 1 above)
Comment (davisagli) [reply metadeploy/api/admin.py]: @jgerigmeyer you're thinking about the API that supports the frontend, but there is also the separate admin API that we use for publishing new plan versions: https://github.com/oddbird/MetaDeploy/blob/main/metadeploy/adminapi/api.py#L50

Comment (jgerigmeyer) [reply metadeploy/api/admin.py]: @davisagli Ah, makes sense. I can add the same validation to the serializer(s) then.

Comment (jgerigmeyer) [reply metadeploy/api/admin.py]: @davisagli Added validation to both serializers. Would you prefer a shared helper fn rather than duplicated?

**Review by davisagli — APPROVED** [review]

**Pull request merged**